pkg/global/apply_configuration.go

```diff
@@ -70,9 +42,20 @@ func ApplyConfiguration(configuration *pb.Configuration) error {
 			view.RegisterExporter(pe)
 		}
 
-		if tracingConfiguration.AlwaysSample {
-			trace.ApplyConfig(trace.Config{DefaultSampler: trace.AlwaysSample()})
+		sampling := float64(tracingConfiguration.SampleProbability)
+		if otelConfiguration := tracingConfiguration.GetOtel(); otelConfiguration != nil {
+			if err := applyOtel(otelConfiguration, sampling); err != nil {
+				return err
+			}
 		}
+
+		if ocConfiguration := tracingConfiguration.GetOc(); ocConfiguration != nil {
+			if err := applyOc(ocConfiguration); err != nil {
+				return err
+			}
+		}
+
+		octrace.ApplyConfig(octrace.Config{DefaultSampler: octrace.ProbabilitySampler(float64(tracingConfiguration.SampleProbability))})
```

Comment:
I guess this casting could be removed, if the Protobuf configuration used `double` instead of `float`?

Why do we need to call into octrace here? Isn't this code supposed to be decoupled from OpenCensus/OpenTelemetry?

Reply:
Ah yes, protobuf does have a double type.

It doesn't matter much, but I moved this into applyOpenTelemetry. There's no semantic difference.